src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp

```diff
@@ -634,8 +644,17 @@ void runtime::intelgpu::do_logic_kernel(cldnn::topology& topology,
             ++var_idx;
         }
 
-        writer << "if (inputA" << access_dims(inputA_shape) << operation << "inputB"
-               << access_dims(inputB_shape) << ")\n";
+        if (operation == " && " || operation == " || ")
+        {
+            writer << "if (inputA" << access_dims(inputA_shape) << "!= 0" << operation << "inputB"
```

Comment:
I don't think you have to use `!=0` here. At least after input/output data types changed to `bool`. And you will not have to use additional `if` above.